lib/fs/__tests__/realpath.test.js

```diff
@@ -1,11 +1,72 @@
 'use strict'
 
-const fse = require('../..')
+const fs = require('fs')
+const path = require('path')
 const assert = require('assert')
 
 /* eslint-env mocha */
 
+describe('realpath.native does not exist', () => {
+  let warning
+  const originalListener = process.listeners('warning')[process.listeners('warning').length - 1]
+  const warningListener = error => {
+    if (error.name === 'Warning') {
+      if (error.code.startsWith('fs-extra-WARN0003')) {
+        warning = error
+        return
+      }
+    }
+    originalListener(error)
+  }
+
+  const realpathNativeBackup = fs.realpath.native
+  const clearFseCache = () => {
+    const fsePath = path.dirname(require.resolve('../../..'))
+    for (const entry in require.cache) {
+      if (entry.startsWith(fsePath)) {
+        delete require.cache[entry]
+      }
+    }
+  }
+
+  before(() => {
+    process.off('warning', originalListener)
```

Comment:
I don't mind the warning being printed during tests; I'd rather leave the `originalListener` intact.

Reply:
Noted. Now uses warningListener while leaving originalListener intact.